src/v/datalake/translation/partition_translator.cc

```diff
@@ -108,7 +123,7 @@ partition_translator::partition_translator(
   // TODO: type resolver and record translator should be constructed based on
   // topic configs.
   , _type_resolver(type_resolver)
-  , _record_translator(std::make_unique<default_translator>())
+  , _record_translator(make_record_translator(mode))
```

Comment:
Also need to update the `type_resolver` to switch between `record_schema_resolver` and `binary_type_resolver`. As is, we're still parsing magic bytes from the record even when in key-value mode

Reply:
Ah good catch.. actually I assumed this logic is all wrapped into the translator .. meaning `key_value_translator` picks `binary_type_resolver` and `structured_translator` picks `record_schema_resolver`, any reason thats not the case?

Reply:
No reason, they're just separate abstractions without a strong dependence on one another. You can imagine in the future a `type_resolver` that resolves the schema via schema subject, but still using the `structured_translator`